### PR TITLE
[Fix #38] Add redirect for /api-docs ...

### LINKED ISSUE #38: /api-docs not working as expected
Make calls to `/api-docs` be redirected to `/api-docs/index.html` in order to properly render swagger assets.

### PR DESCRIPTION
**Pull request by harenson**
… and:
- split cowboy_swagger_handler in two more handlers
- update coverspec

### CODE REVIEW

File: src/cowboy_swagger.erl [hunk 1]
```diff
@@ -92,11 +92,8 @@ validate_metadata(Metadata) ->
 -spec filter_cowboy_swagger_handler([trails:trail()]) -> [trails:trail()].
 filter_cowboy_swagger_handler(Trails) ->
```
Comment (gadgetci) [inline src/cowboy_swagger.erl]: According to **Xref**:
> `cowboy_swagger:filter_cowboy_swagger_handler/1` calls undefined function `maps:filter/2`

Comment (elbrujohalcon) [inline src/cowboy_swagger.erl]: @gadgetci you really should start using Erlang/OTP 18.x

File: src/cowboy_swagger.erl [hunk 2]
```diff
@@ -92,11 +92,8 @@ validate_metadata(Metadata) ->
...
   F = fun(Trail) ->
-    case trails:handler(Trail) of
-      cowboy_swagger_handler  -> false;
-      cowboy_static           -> false;
-      _                       -> true
-    end
+    MD = trails:metadata(Trail),
+    maps:size(maps:filter(fun is_visible/2, MD)) /= 0
```
Comment (gadgetci) [inline src/cowboy_swagger.erl]: According to **Dialyzer**:
>  Call to missing or unexported function maps:filter/2

File: src/cowboy_swagger_redirect_handler.erl [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  {upgrade, protocol, cowboy_rest}.
+
+%% @hidden
+-spec rest_init(cowboy_req:req(), options()) ->
+  {ok, cowboy_req:req(), options()}.
+rest_init(Req, Opts) ->
+  {ok, Req, Opts}.
+
+%% @hidden
+-spec content_types_provided(cowboy_req:req(), state()) ->
```
Comment (elbrujohalcon) [inline src/cowboy_swagger_redirect_handler.erl]: This function is probably useless. Can you try removing it to see if everything still works, please?

File: src/cowboy_swagger_redirect_handler.erl [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+%% @hidden
+-spec init({atom(), atom()}, cowboy_req:req(), options()) ->
+  {upgrade, protocol, cowboy_rest}.
+init(_Transport, _Req, _Opts) ->
+  {upgrade, protocol, cowboy_rest}.
+
+%% @hidden
+-spec rest_init(cowboy_req:req(), options()) ->
+  {ok, cowboy_req:req(), options()}.
+rest_init(Req, Opts) ->
```
Comment (elbrujohalcon) [inline src/cowboy_swagger_redirect_handler.erl]: You can safely remove this function

**Pull request merged**